packages/peregrine/lib/talons/CheckoutPage/__tests__/useCheckoutPage.spec.js

```diff
@@ -0,0 +1,358 @@
+
+const props = {
+    mutations: { createCartMutation, placeOrderMutation },
+    queries: { getCheckoutDetailsQuery, getOrderDetailsQuery }
+};
+
+const readQuery = jest.fn();
+const writeQuery = jest.fn();
+const client = { readQuery, writeQuery };
+
+const getOrderDetails = jest.fn();
+const getOrderDetailsQueryResult = jest.fn().mockReturnValue([
+    getOrderDetails,
+    {
+        data: null,
+        loading: false,
+        called: false
+    }
+]);
+
+const getCheckoutDetailsQueryResult = jest.fn().mockReturnValue({
+    data: null,
+    networkStatus: 0
+});
+
+const createCart = jest.fn();
+const createCartMutationResult = jest.fn().mockReturnValue([
+    createCart,
+    {
+        error: null,
+        loading: false,
+        data: null
+    }
+]);
+const placeOrder = jest.fn();
+const placeOrderMutationResult = jest.fn().mockReturnValue([
+    placeOrder,
+    {
+        error: null,
+        loading: false,
+        data: null
+    }
+]);
+
+/**
+ * Helpers
+ */
+
+const Component = props => {
+    const talonProps = useCheckoutPage(props);
+
+    return <i talonProps={talonProps} />;
+};
+
+const getTalonProps = props => {
+    const tree = createTestInstance(<Component {...props} />);
+    const { root } = tree;
+    const { talonProps } = root.findByType('i').props;
+
+    const update = newProps => {
+        act(() => {
+            tree.update(<Component {...{ ...props, ...newProps }} />);
+        });
+
+        return root.findByType('i').props.talonProps;
+    };
+
+    return { talonProps, tree, update };
+};
+
+/**
+ * beforeAll
+ */
+
+beforeAll(() => {
+    useQuery.mockImplementation(query => {
+        if (query === getCheckoutDetailsQuery) {
+            return getCheckoutDetailsQueryResult();
+        } else {
+            return {};
+        }
+    });
+
+    useLazyQuery.mockImplementation(query => {
+        if (query === getOrderDetailsQuery) {
+            return getOrderDetailsQueryResult();
+        } else {
+            return [jest.fn(), {}];
+        }
+    });
+
+    useMutation.mockImplementation(mutation => {
+        if (mutation === createCartMutation) {
+            return createCartMutationResult();
+        } else if (mutation === placeOrderMutation) {
+            return placeOrderMutationResult();
+        } else {
+            return [jest.fn(), {}];
+        }
+    });
+
+    useApolloClient.mockReturnValue(client);
+});
+
+/**
+ * Tests
+ */
+
+test('Should return correct shape', () => {
+    const { talonProps } = getTalonProps(props);
+
+    expect(talonProps).toMatchSnapshot();
+});
+
+test('isLoading should be set to true if the checkout details query networkStatus is less than 7', () => {
+    getCheckoutDetailsQueryResult.mockReturnValueOnce({
+        data: null,
+        networkStatus: 4
+    });
+
+    const { talonProps, update } = getTalonProps(props);
+
+    expect(talonProps.isLoading).toBeTruthy();
+
+    getCheckoutDetailsQueryResult.mockReturnValueOnce({
```

Comment:
Starting from this line down should be its own separate test. "isLoading should be set to false if the checkout details query networkStatus is greater than or equal to 7".

I'd also test seven (7) explicitly since that's where the errors will likely occur.